src/common/types/vector.cpp

```diff
@@ -1083,6 +1084,9 @@ void Vector::Verify(Vector &vector_p, const SelectionVector &sel_p, idx_t count)
 				}
 			}
 		}
+		if (vector->GetType().id() == LogicalTypeId::MAP) {
+			VerifyMap(*vector, count);
```

Comment:
You will either need to use the `sel` input to the `Verify` function here, or move this check up. Check how dictionaries are handled at line 1005. Alternatively, move this to the other `Vector::Verify` function (that does not have a selection vector as argument).

Reply:
So if I understand correctly, with a selection vector we can essentially mask out rows that shouldn't be checked, a Dictionary Vector is essentially just a wrapper around a vector with an added selection vector, so in the case of a dictionary vector we want to merge these selection vectors.

This sounds like the cause of the issue I was having yesterday evening, where I have an entry that is NULL, where count=1, but row0 and row1 are valid, only row2 isn't valid

sounds like the dictionary vector in this case has a selection vector that maps index 0 -> index 2

Reply:
This appears to work, but I'm not quite sure I understand
```c++
	for (idx_t row = 0; row < count; row++) {
		auto mapped_row = sel.get_index(row);
		auto row_idx = vdata.sel->get_index(mapped_row);
		// map is allowed to be NULL
		if (!map_validity.RowIsValid(row_idx)) {
			continue;
		}
		row_idx = key_vdata.sel->get_index(row);
```
I first thought I'd want to continue using the mapped_row on the child vector as well
`row_idx = key_vdata.sel->get_index(mapped_row);`
Because this would relate to the same row index as I just checked the validity for ?